### ISSUE TITLE
[Feature Request] Provide the option to manage the jquery libraries

### ISSUE THREAD

**Issue by naveenvalecha** [acquia/blt#525]
Right now there's only option to inlucde the libaries is inside the docroot/libraries folder but BLT excludes the [docroot/libraries](https://github.com/acquia/blt/blob/8.x/template/.gitignore#L17) directory.

Libraries API module which provides the option to manage the libraries which is not much better. So please suggest the better approach for managing the jquery libraries.

Thanks
Naveen

**Comment by grasmash** [comment]
I don't believe that managing jQuery directly is within the scope of BLT. However, I think you could rephrase this as a support request "How do I deploy frontend librares?" The answer to this question would depend on whether you 1) Are deploying frontend libraries in a custom theme, or 2) Are deploying frontend libraries required for a contributed project.

1) **Deploying frontend libraries in a custom theme**. In this instance I would suggest choosing one of the following options
  * Load the library as an external library. See (Adding stylesheets (CSS) and JavaScript (JS) to a Drupal 8 module)[https://www.drupal.org/developing/api/8/assets].
  * Use a package manager

**Comment by grasmash** [comment]
I don't believe that managing jQuery directly is within the scope of BLT.

However, I think you could rephrase this as a support request "How do I deploy frontend librares?" The answer to this question would depend on whether you are 1) deploying frontend libraries in a custom theme, or 2) deploying frontend libraries required for a contributed project.

1. **Deploying frontend libraries in a custom theme**. In this instance I would suggest choosing one of the following options
  * Load the library as an external library. See (Adding stylesheets (CSS) and JavaScript (JS) to a Drupal 8 module)[https://www.drupal.org/developing/api/8/assets].
  * Use a package manager (e.g., [NPM](https://www.npmjs.com/)) to download your dependencies. Then use BLT's `frontend-build` and `post-deploy-build` target-hooks to trigger building those dependencies. E.g., call `npm install` in your theme directory via these hooks.
1. **Deploying frontend libraries required for a contributed project**. In this instance, the contributed project should provide the ability to download and discover the libraries. 
    * Patch the project to follow one of these strategies:
        * Use a command to download the dependency and incorporate that into command  BLT's  `frontend-build` and `post-deploy-build` target-hooks. E.g, 
            * `npm install`
            * a drush command
        * Use the project's composer.json with a [custom repository](https://getcomposer.org/doc/05-repositories.md). I do not recommend this for non PHP and non-binary projects. Composer is designed for PHP projects and it facilitates project discovery using [autoloading](https://getcomposer.org/doc/01-basic-usage.md#autoloading). If you do use composer, ensure that your project attempts to discover the dependency in the appropriate `vendor` subdirectory. I do not recommend using a symlink to place a reference to it. This will inevitably cause complications with virtual machines, deployments, varying host operating systems (e.g., Windows), etc.
    * If you can't patch the module to support one of these strategies, then simply commit the dependency. You can use a custom .gitignore file for you project, [ensure that it is copied to the deployment artifact, and supply your own, custom .gitignore file to be used in the deployment artifact](http://blt.readthedocs.io/en/8.x/readme/extending-blt/#deploybuild).

**Comment by grasmash** [comment]
Documentation has been added here: https://github.com/acquia/blt/pull/527
